crates/ruff_python_formatter/src/pattern/pattern_match_sequence.rs

```diff
@@ -51,3 +43,54 @@ impl FormatNodeRule<PatternMatchSequence> for FormatPatternMatchSequence {
         }
     }
 }
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub(crate) enum SequenceType {
+    /// A list literal, e.g., `[1, 2, 3]`.
+    List,
+    /// A parenthesized tuple literal, e.g., `(1, 2, 3)`.
+    Tuple,
+    /// A tuple literal without parentheses, e.g., `1, 2, 3`.
+    TupleNoParens,
+}
+
+impl SequenceType {
+    pub(crate) fn from_pattern(pattern: &PatternMatchSequence, source: &str) -> SequenceType {
+        if source[pattern.range()].starts_with('[') {
+            SequenceType::List
+        } else if source[pattern.range()].starts_with('(') {
+            let Some(elt) = pattern.patterns.first() else {
+                return SequenceType::Tuple;
+            };
+
+            // Count the number of open parentheses between the start of the pattern and the first element.
+            let open_parentheses_count =
+                SimpleTokenizer::new(source, TextRange::new(pattern.start(), elt.start()))
+                    .skip_trivia()
+                    .filter(|token| token.kind() == SimpleTokenKind::LParen)
+                    .count();
+
+            // Count the number of parentheses between the end of the first element and its trailing comma.
+            let close_parentheses_count =
+                SimpleTokenizer::new(source, TextRange::new(elt.end(), elt.end()))
+                    .skip_trivia()
+                    .take_while(|token| token.kind() != SimpleTokenKind::Comma)
+                    .filter(|token| token.kind() == SimpleTokenKind::RParen)
+                    .count();
```

Comment:
I'm not able to follow this. Can you add an example/test cases for the 2 cases here?